### PR TITLE
Telegram chain support (learn-in exercise)

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Added basic support for Telegram TON blockchain, basic infrastructure and address handling.

## Testing instructions

Unit tests expanded.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Finalize coin ID number, also introduce to SLIP44 registry
- [ ] Add TON logo to trustwallet/assets .
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

File: docs/coins.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ This list is generated from [./coins.json](../coins.json)
 | 425     | Aion             | AION   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/aion/info/logo.png" width="32" />         | <https://aion.network>        |
 | 457     | Aeternity        | AE     | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/aeternity/info/logo.png" width="32" />    | <https://aeternity.com>       |
 | 500     | Theta            | THETA  | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/theta/info/logo.png" width="32" />        | <https://www.thetatoken.org>  |
+| 505     | Telegram         | TON    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ton/info/logo.png" width="32" />          | <https://github.com/copperbits/TON> |
```
Comment (kolya182) [inline docs/coins.md]: @catenocrypt Make sure to add coin to SLIP-44 registry https://github.com/satoshilabs/slips/blob/master/slip-0044.md

Comment (catenocrypt) [reply docs/coins.md]: @kolya182 , you mean I should create a PR there?  What value should I chose? I've just picked a random value...

Comment (catenocrypt) [reply docs/coins.md]: SLIP-44 request created,  https://github.com/satoshilabs/slips/pull/752

Comment (kolya182) [reply docs/coins.md]: > @kolya182 , you mean I should create a PR there? What value should I chose? I've just picked a random value...

You picked correct one. Once it merged please make PR to https://github.com/trustwallet/assets with TON chain logo.

Comment (catenocrypt) [reply docs/coins.md]: Slips PR has been merged, TON is included in Slip-44, with value 396.  This PR has been updated to use that value.

**Comment by catenocrypt** [comment]
Implemented TON Cell structures, and new address creation.

**Comment by hewigovens** [comment]
let's track it here: https://github.com/trustwallet/wallet-core/pull/691

**Pull request closed without merging**